File: frontend/the_eventors/lib/models/dto/CommentResponseDto.dart
import 'dart:convert';
import 'dart:io';

import 'package:http/http.dart';

CommentResponseDto newsModelFromJson(String str) =>
    CommentResponseDto.fromJson(json.decode(str));

String newsModelToJson(CommentResponseDto data) => json.encode(data.toJson());

class CommentResponseDto {
  int id;
  String message;
  String createdAt;
  String username;
  String profileImage;
  int idEvent;

  List<CommentResponseDto> replies;

  CommentResponseDto({
    required this.id,
    required this.message,
    required this.createdAt,
    required this.username,
    required this.profileImage,
    required this.idEvent,
    required this.replies,
  });

  factory CommentResponseDto.fromJson(Map<String, dynamic> json) =>
      CommentResponseDto(
          id: json["id"],
          message: json["message"],
          createdAt: json["createdAt"],
          username: json["username"],
          profileImage: json["profileImage"],
          idEvent: json["idEvent"],
          replies: json["replies"]);

  Map<String, dynamic> toJson() => {
        "id": id,
        "message": message,
        "createdAt": createdAt,
        "username": username,
        "profileImage": profileImage,
        "idEvent": idEvent,
        "replies": replies,
      };
}

File: backend/the-eventors/src/main/java/com/example/theeventors/model/dto/CommentResponseDto.java
package com.example.theeventors.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CommentResponseDto {

    Long id;
    String message;
    LocalDateTime createdAt;
    String username;

    String profileImage;
    Long idEvent;

    List<CommentResponseDto> replies;

    public CommentResponseDto(Long id, String message, LocalDateTime createdAt, String username,String profileImage, Long idEvent) {
        this.id = id;
        this.message = message;
        this.createdAt = createdAt;
        this.username = username;
        this.profileImage = profileImage;
        this.idEvent = idEvent;
        this.replies = new ArrayList<>();
    }
}
